src/provider/providerWrapper.ts

```diff
@@ -135,9 +134,9 @@ export class ProviderWrapper {
     signature: string,
   ): Promise<string> {
     if (this.type === ProviderTypes.ETHEREUM) {
-      const encodedParams = encodeParameters(
-        ['bytes32', 'bytes'],
-        [hash, signature],
+      const encodedParams = encodeAbiParameters(
+        [{ type: 'bytes32' }, { type: 'bytes' }],
+        [hash as Hex, signature as Hex],
```

Comment:
Can't we replace the function param type to `0x${string}` as well here? (or `Hex`) so that we can improve the typing and remove all of these `as` inside the function?

```ts
  async isValidSignature(
    address: string,
    hash: Hex,
    signature: Hex,
```

Reply:
There are some good types in viems that can be imported. But not sure how much it would be breaking for applications.

https://viem.sh/docs/glossary/types.html#hash

<img width="283" alt="image" src="https://github.com/user-attachments/assets/a95a04f2-0508-45a1-bfc6-40844ff0a9e2" />